### ISSUE TITLE
Allow Data URLs in offlineRegionDefinitions styleURL

### ISSUE THREAD

**Issue by mariusvn** [maplibre/maplibre-gl-native#609]
**Is your feature request related to a problem? Please describe.**
The offline region definition require an URL to define the style of the downloaded data. This makes the download of custom style generated on the fly impossible.

**Describe the solution you'd like**
Would be nice to allow raw json or data URL that contains a style definition.

**Describe alternatives you've considered**
As i said before, I tried with the data URLs in the style URL but it doesn't seems like the URL parser parse it.

**Additional context**
MapLibre gives me this message followed by the content of the styleURL: `[HTTP] Unable to parse resourceUrl`

**Comment by boldtrn** [comment]
Yes, I think we should allow to pass a Style.Builder to the offline regions.

**Comment by mariusvn** [comment]
yeah that would be even better

**Comment by louwers** [comment]
This is apparently working on iOS (using a `file://` URL), but not on Android. This error is coming from here:

https://github.com/maplibre/maplibre-native/blob/d15de921581ff500a48d1a463d33902bd499fad7/platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/module/http/HttpRequestImpl.java#L65

So apparently this hits the 'HTTP file source' instead of the 'Local File Source' implementation. Maybe fixing this is as easy as changing

https://github.com/maplibre/maplibre-native/blob/d15de921581ff500a48d1a463d33902bd499fad7/src/mbgl/storage/http_file_source.hpp#L17-L19

to return `false` when the URL scheme of the requested resource is `file://`, i.e. the inverse of

https://github.com/maplibre/maplibre-native/blob/d15de921581ff500a48d1a463d33902bd499fad7/platform/default/src/mbgl/storage/local_file_source.cpp#L15-L17

**Comment by ccannell** [comment]
Interested in a fix. This is a blocker for an app I'm writing for a client.

**Comment by mariusvn** [comment]
it may be an error that came with android 13 new storage restrictions

**Comment by sargunv** [comment]
+1 to allowing style builders in the offline manager; this would allow dynamic styles like those created in Compose to work with it too. 

And on iOS, I guess something similar to the snapshotter (exposing style for mutation?)
